docs/receiving/verifying-payloads/how.mdx

```diff
@@ -210,32 +210,294 @@ No easy way to verify the signature just with cURL.
+}
+```
+
+### Go (Gin)
+
+```go
+package main
+
+import (
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	svix "github.com/svix/svix-libs/go"
+)
+
+const secret = "MfKQ9r8GKYqrTwjUPD8ILPZIo2LaLaSw"
+
+func main() {
+
+	wh, err := svix.NewWebhook(secret)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	r := gin.Default()
+	r.POST("/webhook", func(c *gin.Context) {
+		headers := c.Request.Header
+		payload, err := io.ReadAll(c.Request.Body)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		err = wh.Verify(payload, headers)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		// Do something with the message...
+
+		c.JSON(200, gin.H{})
+	})
+	r.Run()
+}
+```
+
+### Ruby (Ruby on Rails)
+
+Once you've setup your project add a route to your `config/routes.rb` file at the top of the `Rails.application.routes.draw` block:
+
+```rb
+Rails.application.routes.draw do
+  post "/webhook", to: "webhook#index"
+
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+end
+```
+
+The route above declares that `POST /webhook` requests are mapped to the index action of `WebhookController`.
+
+To create `WebhookController` and its index action, we'll run the controller generator (with the `--skip-routes` option because we already have an appropriate route):
+
+```sh
+bin/rails generate controller Webhook index --skip-routes
+```
+
+Rails will create several files for you:
+
+```sh
+    create  app/controllers/webhook_controller.rb
+    invoke  erb
+    create    app/views/webhook
+    create    app/views/webhook/index.html.erb
+    invoke  test_unit
+    create    test/controllers/webhook_controller_test.rb
+    invoke  helper
+    create    app/helpers/webhook_helper.rb
+    invoke    test_unit
+    invoke  assets
+    invoke    scss
+    create      app/assets/stylesheets/webhook.scss
+```
+
+Now we can add our verification logic to the newly created `app/controllers/webhook_controller.rb` file:
+
+```rb
+require 'svix'
+
+class WebhookController < ApplicationController
+  protect_from_forgery with: :null_session # disables CSRF middleware; required for API endpoints
+
+  def index
+    begin
+      payload = request.body.read
+      headers = request.headers
+      wh = Svix::Webhook.new("MfKQ9r8GKYqrTwjUPD8ILPZIo2LaLaSw")
+
+      json = wh.verify(payload, headers)
+
+      # Do something with the message...
+
+      head :no_content
+    rescue
+      head :bad_request
+    end
+  end
+end
+```
+
+### PHP (Laravel)
+
+In your `routes/api.php` file add the following after the last use directive:
+
+```php
+use Svix\Webhook;
+use Svix\Exception\WebhookVerificationException;
+
+Route::post('webhook', function(Request $request) {
+    $payload = $request->getContent();
+    $headers = collect($request->headers->all())->transform(function ($item) {
+        return $item[0];
+    });
```

Comment:
Apache, and libCurl return headers as an array of `string -> string` which is what our library expects. Laravel however, returns headers as an array of `string -> array[string]` so this transform is required.  Gonna see if it's easy enough to change the PHP library to accept either.

Reply:
Let's not support it in the lib for now.